sdk/storage/examples/data_lake_00.rs

```diff
@@ -30,11 +30,17 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
     println!("token expires on {}", bearer_token.expires_on);
     println!();
 
-    let data_lake = storage_account_client
-        .as_storage_client()
-        .as_data_lake_client(account, bearer_token.token.secret().to_owned())?;
+    let storage_client = storage_account_client.as_storage_client();
+    let data_lake = DataLakeClient::new(
+        storage_client,
+        account,
+        bearer_token.token.secret().to_owned(),
+        None,
+    );
 
-    let file_system = data_lake.as_file_system_client(&file_system_name)?;
+    let file_system = data_lake
+        .clone()
```

Comment:
Is there a better way than cloning the DataLakeClient here? This stems from `Box::pin(data_lake...` below.

Reply:
Not right now, but we want cloning to be cheap so this shouldn't be an issue. I've talked with @yoshuawuyts about making this more ergonomic. The start of this conversation is [here](https://github.com/Azure/azure-sdk-for-rust/pull/474#issuecomment-959525371).